patroni/ha.py

```diff
@@ -121,6 +122,15 @@ def bootstrap(self):
             return 'waiting for leader to bootstrap'
 
     def recover(self):
+        if self.has_lock() and self.update_lock():
+            if self.patroni.config['master_start_timeout'] == 0:
+                # We are requested to prefer failing over to restarting master. But see first if there
+                # is anyone to fail over to.
+                if self.is_failover_possible(self.cluster.members):
```

Comment:
So far `is_failover_possible` method was used only for manual failover and it doesn't take into account xlog location of the nodes. It could happen that all the nodes are "lagging" more than `maximum_lag_on_failover` and will consider themselves  as not healthy enough for the leader race.
I think it is a good time to fix "TODO" on the line 236